### ISSUE TITLE
Too many hangs when a large header set is included (e.g. webkitgtk)

### ISSUE THREAD

**Issue by Ambrevar** [nflath/c-eldoc#14]
If I include a large header set like webkitgtk:

```lisp
(setq c-eldoc-includes "`pkg-config gtk+-3.0  --cflags` -I./ -I../ `pkg-config webkit2gtk-4.0 --cflags`")
```

then every out-of-cache lookups hangs for some seconds.  This makes this package unusable.  I haven't profiled it, but there might be some way to optimize this.

**Comment by nflath** [comment]
Yeah, I suspect this is going to be due to running the preprocessor.  Fixing this would be tons of work.
